src/core/functions/user_functions/UserScriptFunctions.ts

```diff
@@ -66,7 +70,36 @@ export class UserScriptFunctions implements IGenerateObject {
     }
 
     async generate_object(): Promise<Record<string, unknown>> {
-        const user_script_functions = await this.generate_user_script_functions();
+        const user_script_functions =
+            await this.generate_user_script_functions();
         return Object.fromEntries(user_script_functions);
     }
+
+    async inline_requires(file_content: string): Promise<string> {
+        const require = file_content.match(/require\('\.\/(.*)(\.js)?'\)/);
+        if (!require) {
+            return file_content;
+        }
+        if (require && require[1]) {
+            const require_path = `${this.plugin.settings.user_scripts_folder}/${require[1]}.js`;
+            const require_file =
+                this.app.vault.getAbstractFileByPath(require_path);
+            let require_content = await this.app.vault.read(require_file as TFile);
+            require_content = require_content.replace(
+                /module\.exports = .*$/,
+                ""
+            );
+            const exists = file_content.includes(require_content);
+            let inlined = file_content;
+            if (!exists) {
+                inlined = file_content.replace(
+                    `const ${require[1]} = ${require[0]}`,
+                    require_content
+                );
+            } else {
+                inlined = file_content.replace(`const ${require[1]} = ${require[0]}`, "");
+            }
+            return this.inline_requires(inlined);
```

Comment:
I'm running into an infinite loop here, and it seems to be happening because of lines 95-98. 

In my file, I have this:

```javascript
const extractDate = require('./date_utils.js');
```

But lines 95-98 expect it to be this:

```javascript
const date_utils = require('./date_utils.js');
```

Since my line does no look like that, the `file_content.replace` never happens, and then upon calling:

```javascript
return this.inline_requires(inlined);
```
The whole process starts over. It sees the `const extractDate = require('./date_utils.js');` line, tries to inline it, but fails to replace the `require` statement with the inline content because my line does not match up with expectations.